src/libraries/Common/src/Internal/Cryptography/SymmetricPadding.cs

```diff
@@ -0,0 +1,220 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Security.Cryptography;
+
+namespace Internal.Cryptography
+{
+    internal static class SymmetricPadding
+    {
+        public static int GetCiphertextLength(int plaintextLength, int paddingSizeInBytes, PaddingMode paddingMode)
+        {
+            Debug.Assert(plaintextLength >= 0);
+
+             //divisor and factor are same and won't overflow.
+            int wholeBlocks = Math.DivRem(plaintextLength, paddingSizeInBytes, out int remainder) * paddingSizeInBytes;
+
+            switch (paddingMode)
+            {
+                case PaddingMode.None when (remainder != 0):
+                    throw new CryptographicException(SR.Cryptography_PartialBlock);
+                case PaddingMode.None:
+                case PaddingMode.Zeros when (remainder == 0):
+                    return plaintextLength;
+                case PaddingMode.Zeros:
+                case PaddingMode.PKCS7:
+                case PaddingMode.ANSIX923:
+                case PaddingMode.ISO10126:
+                    return checked(wholeBlocks + paddingSizeInBytes);
+                default:
+                    Debug.Fail($"Unknown padding mode {paddingMode}.");
+                    throw new CryptographicException(SR.Cryptography_UnknownPaddingMode);
+            }
+        }
+
+        public static int PadBlock(ReadOnlySpan<byte> block, Span<byte> destination, int paddingSizeInBytes, PaddingMode paddingMode)
```

Comment:
FWIW, I (personally) strongly prefer that methods inside internal classes are marked internal, unless they are conforming to an interface.

Every time I see "public" I have an API Review reaction (e.g. something like "is this the right shape for the rest of time?", "did we review this?", or "I don't remember seeing the ref.cs update").

I wouldn't spin the change at this point just for this... but if it has to be spun.... :smile:.

Reply:
Hmm. I suspect in two years you will have changed my mind on this, much like with `var` when I set out to start contributing here a few years ago 😄.